### PR TITLE
[asr] optimize attention cache used for attention ; 0-dim tensor for model export

### PR DESCRIPTION
**Pull request by zh794390558**
### PR types

 Performance optimization
### PR changes

Models
### Describe

1. 优化后相比优化前相对提升：9.9%

conformer_wenetspeech | - | python | attention_rescoring | 0.5 | 16 | -1 | 0.8712686765174202(utts=40) | -
-- | -- | -- | -- | -- | -- | -- | -- | --
conformer_wenetspeech（reduce attention cache） | - | python | attention_rescoring | 0.5 | 16 | -1 | 0.7847489114800089(utts=40) |  

2. using 0-dim tensor (tensor(0,0,0,0) )  to simple `to_static` conformer model export
2. fix cli to support scp batch process
3. add pybind11 to install deps.

### CODE REVIEW

File: paddlespeech/s2t/modules/encoder.py [hunk 1]
```diff
@@ -225,50 +226,50 @@ def forward_chunk(
...
+        # before embed, xs=(B, T, D1), pos_emb=(B=1, T, D)
+        xs, pos_emb, _ = self.embed(xs, tmp_masks, offset=offset) 
+        # after embed, xs=(B=1, chunk_size, hidden-dim)
 
-        if subsampling_cache is not None:
-            cache_size = subsampling_cache.shape[1]  #T
-            xs = paddle.cat((subsampling_cache, xs), dim=1)
-        else:
-            cache_size = 0
+        elayers = paddle.shape(att_cache)[0]
```
Comment (Jackwaterveg) [inline paddlespeech/s2t/modules/encoder.py:233]: subsampling_cache 不用了？

Comment (Jackwaterveg) [reply paddlespeech/s2t/modules/encoder.py:233]: 缓存的计算量确实不大，不过可能也可以节省一定时间

Comment (zh794390558) [reply paddlespeech/s2t/modules/encoder.py:233]: 合并到att_cache里了，会降低一些计算。

Comment (Jackwaterveg) [reply paddlespeech/s2t/modules/encoder.py:233]: ok，理解了

**Review by Jackwaterveg — APPROVED** [review]
LGTM

**Review by Jackwaterveg — APPROVED** [review]
LGTM

**Pull request merged**